subscribe.go

```diff
@@ -0,0 +1,71 @@
+package bps
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"sync"
+)
+
+var (
+	subReg   = map[string]SubscriberFactory{}
+	subRegMu sync.Mutex
+)
+
+// SubscribeOptions defines subscribe options.
+type SubscribeOptions struct {
+	// BatchSize holds desired message batch size.
+	// Default: 1.
+	BatchSize int
+}
+
+// ----------------------------------------------------------------------------
+
+// Subscriber defines the main subscriber interface.
+type Subscriber interface {
+	// Subscribe subscribes for topic messages and blocks till context is cancelled or error occurs.
+	//
+	// Handler is provided an atomic message batch (entire batch either succeeds or fails).
+	// Batch is guaranteed to contain at least 1 message.
+	Subscribe(
+		ctx context.Context,
+		topic string,
+		handler func([]Message) error,
```

Comment:
`[]Message` is not an overkill - I'd like to make atomic-batch a base approach for consuming. Probably, we can have simpler `Subscriber` wrapper with single-message handler.

I did not extract handler func type not to scroll the docs when using bps.Consumer for it.

Reply:
let's add a `Handler` interface with `ProcessMessage(contenxt.Context, Msg) error` and a `HandlerFunc`. We should also not operate on `Message` struct but on an `Msg` interface. One is for producing the other for consuming.

Reply:
@dim maybe better this?

```go
type Message interface { ... }
type ProduceMessage struct { ... }
```

Msg/Message are kinda confusing / don't tell much (IMO).

Reply:
I'm doing this ^^